### PR TITLE
Fix error messages for payment class and refactored code

### LINKED ISSUE #147: [PE-D] Doing `payment receive/` with a receive date that is in the future results in an unexpected `payment by/` date being set.
Doing `payment receive/19-02-2022` without having set a `payment by/` date makes the `payment by/` date be auto-set to `19-02-2022` which does not make sense from a feature standpoint.

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: CraveToCode/ped#10

### PR DESCRIPTION
**Pull request by ykwei7**
Resolved #147, #126

For #147, changed payment message from `Overdue: No (By: XXX)` to `Overdue: No (Next payment date by: XXX)` to make it more explicit that it refers to the next upcoming payment date to be expected.

For #126, adjusted according to what was suggested and changed error message to `tutee index is invalid`.

Added test cases for payment commands as well

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/paymentcommand/PaymentReceiveCommand.java [hunk 1]
```diff
@@ -71,23 +75,25 @@ public CommandResult execute(Model model) throws CommandException {
         Payment existingPayment = tuteeToGet.getPayment();
         String existingPaymentValue = existingPayment.getValue();
         LocalDate existingPayByDate = existingPayment.getPayByDate();
-        Tutee editedTutee;
 
-        if (zeroPaymentVal.equals(existingPaymentValue) && newPayByDate == null && existingPayByDate == null) {
+        boolean hasZeroPaymentValue = ZERO_PAYMENT_VAL.equals(existingPaymentValue);
```
Comment (atyhamos) [inline src/main/java/seedu/address/logic/commands/paymentcommand/PaymentReceiveCommand.java:79]: As "0" is not equal to "0.00", perhaps a simpler implementation would be 
```suggestion
        boolean hasZeroPaymentValue = Double.parseDouble(existingPaymentValue) == 0;
```

File: src/main/java/seedu/address/model/tutee/Payment.java [hunk 2]
```diff
@@ -126,7 +126,7 @@ public String getOverdueStatus() {
         } else if (payByDateAsString.equals("-")) {
             return "No (Pay-by date not set)";
         } else {
-            return "No (by " + payByDateAsString + ")";
+            return "No (Next payment date by: " + payByDateAsString + ")";
```
Comment (atyhamos) [inline src/main/java/seedu/address/model/tutee/Payment.java:129]: I think this will clear the confusion that our testers had 😅

**Review by atyhamos — APPROVED** [review]
Good job on the thorough test cases! Left a few comments and suggestions. 😄

File: src/main/java/seedu/address/logic/commands/paymentcommand/PaymentReceiveCommand.java (hunk 1 above)
Comment (ykwei7) [reply src/main/java/seedu/address/logic/commands/paymentcommand/PaymentReceiveCommand.java]: Thanks! Have committed the above suggestions

**Pull request merged**